packages/wasm/crate/src/tx.rs

```diff
@@ -273,7 +276,89 @@ pub async fn transaction_info_inner(
+                    let derived_nullifier_from_swap =
+                        Nullifier::derive(fvk.nullifier_key(), swap_position, &commitment);
+
+                    let transaction_infos = storage.get_transaction_infos().await?;
+
+                    for transaction_info in transaction_infos {
+                        transaction_info
+                            .transaction
+                            .and_then(|transaction| transaction.body)
+                            .map(|body| {
+                                for action in body.actions.iter() {
+                                    let mut should_break = false;
+
+                                    action.action.as_ref().and_then(|action_action| {
+                                        if let penumbra_proto::core::transaction::v1::action::Action::SwapClaim(swap_claim) = action_action {
+                                            Some(swap_claim)
+                                        } else {
+                                            None
+                                        }
+                                    }).and_then(|swap_claim| {
+                                        swap_claim.body.as_ref()
+                                    }).map(|body|
+                                        body.nullifier.as_ref().map(|swap_claim_nullifier| {
+                                            if *swap_claim_nullifier == derived_nullifier_from_swap.to_proto() {
+                                                should_break = true;
+
+                                                transaction_info.id.clone()
+                                                    .and_then(|id| {
+                                                        let result = TransactionId::try_from(id);
+                                                        if let Ok(transaction_id) = result {
+                                                            Some(transaction_id)
+                                                        } else {
+                                                            None
+                                                        }
+                                                    }).and_then(|id| {
+                                                        txp.nullification_transaction_ids_by_commitment.insert(
+                                                            commitment,
+                                                            id,
+                                                        )
+                                                    });
+                                            }
+                                        })
+                                    );
+
+                                    if should_break {
+                                        break;
+                                    }
+
+                                }
+
+                                None::<()>
```

Comment:
This feels absurdly unidiomatic, but the linter complained until I returned an `Option` enum from this closure.